iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/OpenHouseInternalCatalog.java

```diff
@@ -128,7 +131,13 @@ public boolean dropTable(TableIdentifier identifier, boolean purge) {
 
   @Override
   public void renameTable(TableIdentifier from, TableIdentifier to) {
-    throw new UnsupportedOperationException("Rename Tables not implemented yet");
+    Table fromTable = loadTable(from);
+    Transaction transaction = fromTable.newTransaction();
+    UpdateProperties updateProperties = transaction.updateProperties();
+    updateProperties.set(CatalogConstants.OPENHOUSE_TABLEID_KEY, to.name());
+    updateProperties.set(CatalogConstants.OPENHOUSE_DATABASEID_KEY, to.namespace().toString());
+    updateProperties.commit();
+    transaction.commitTransaction();
```

Comment:
Changes on the this seems related house tables layer. It seems we are committing properties update as part of OpenhouseInternalRepository. We can decide where do we want to commit the transaction for this. In a way it also makes sense commit the transaction as part of catalog. Also this layer is not part of house table. Shall we make this change as part of next PR along with tables service layer?

Reply:
Sure sounds good